Corona-Warn-App/src/main/java/de/rki/coronawarnapp/storage/SubmissionRepository.kt

```diff
@@ -1,88 +1,86 @@
 import de.rki.coronawarnapp.ui.submission.ApiRequestState
 import de.rki.coronawarnapp.util.DeviceUIState
 import de.rki.coronawarnapp.util.Event
-import de.rki.coronawarnapp.util.di.AppInjector
+import de.rki.coronawarnapp.util.TimeStamper
+import de.rki.coronawarnapp.util.coroutine.AppCoroutineScope
+import de.rki.coronawarnapp.util.coroutine.AppScope
 import de.rki.coronawarnapp.util.formatter.TestResult
 import de.rki.coronawarnapp.worker.BackgroundWorkScheduler
 import kotlinx.coroutines.flow.Flow
 import kotlinx.coroutines.flow.MutableStateFlow
 import kotlinx.coroutines.launch
 import timber.log.Timber
 import java.util.Date
-
-object SubmissionRepository {
-
-    private val appScope by lazy {
-        AppInjector.component.appScope
+import javax.inject.Singleton
+
+@Singleton
+class SubmissionRepository @AssistedInject constructor(
+    @Assisted private val submissionSettings: SubmissionSettings,
+    private val submissionService: SubmissionService,
+    @AppScope val appScope: AppCoroutineScope
+) {
+
+    companion object {
+        fun submissionSuccessful() {
+            BackgroundWorkScheduler.stopWorkScheduler()
+            LocalData.numberOfSuccessfulSubmissions(1)
+        }
+        fun deleteRegistrationToken() {
+            LocalData.registrationToken(null)
+            LocalData.devicePairingSuccessfulTimestamp(0L)
+        }
     }
 
-    val uiStateStateFlowInternal = MutableStateFlow(ApiRequestState.IDLE)
+    private val uiStateErrorInternal = MutableLiveData<Event<CwaWebException>>(null)
+    val uiStateError: LiveData<Event<CwaWebException>> = uiStateErrorInternal
+
+    private val uiStateStateFlowInternal = MutableStateFlow(ApiRequestState.IDLE)
     val uiStateStateFlow: Flow<ApiRequestState> = uiStateStateFlowInternal
-    val uiStateState: LiveData<ApiRequestState> = uiStateStateFlow.asLiveData()
 
     private val testResultReceivedDateFlowInternal = MutableStateFlow(Date())
     val testResultReceivedDateFlow: Flow<Date> = testResultReceivedDateFlowInternal
 
     private val deviceUIStateFlowInternal = MutableStateFlow(DeviceUIState.UNPAIRED)
     val deviceUIStateFlow: Flow<DeviceUIState> = deviceUIStateFlowInternal
 
-    private val testResultFlow = MutableStateFlow<TestResult?>(null)
+    // to be used by new submission flow screens
+    val hasGivenConsentToSubmission = submissionSettings.hasGivenConsent.flow.asLiveData()
 
-    private suspend fun fetchTestResult(): DeviceUIState = try {
-        val testResult = SubmissionService.asyncRequestTestResult()
-        updateTestResult(testResult)
-        deriveUiState(testResult)
-    } catch (err: NoRegistrationTokenSetException) {
-        DeviceUIState.UNPAIRED
-    }
-
-    fun updateTestResult(testResult: TestResult) {
-        this.testResultFlow.value = testResult
-
-        if (testResult == TestResult.POSITIVE) {
-            LocalData.isAllowedToSubmitDiagnosisKeys(true)
-        }
+    private val timeStamper: TimeStamper
```

Comment:
Inject TimperStamper, the idea is to be able to provide a mock one in unit tests so that we have control over time.

Reply:
done